### ISSUE TITLE
Wrong management of sequences and values 

### ISSUE THREAD

**Issue by andrea-narciso** [commandlineparser/commandline#454]
Consider following definition of options:
```
    class Options
    {
        [Option('c', "channels", Required = true, Separator = ':', HelpText = "Channel names")]
        public IEnumerable<string> Channels { get; set; }

        [Value(0,
               Required = true,
               MetaName = "file_path",
               HelpText = "Path of archive to be processed")]
        public string ArchivePath { get; set; }
    }
```
When invoking `app.exe -c chanA:chanB file.hdf5` the expected result would be 
```
Channels = {"chanA", "chanB"} 
ArchivePath = "file.hdf5"
```
Instead, I get a `CommandLine.MissingRequiredOptionError`

If I remove the `Required` attribute from the value definition, parsing succeeds but I get:
`Channels = {"chanA", "chanB", "file.hdf5"} `
And the expected `ArchivePath `value ends up in the `Channels `definitions (why? there is no ":" separator)

Am I doing something wrong?

**Comment by moh-hassan** [comment]
It seems that this bug is similar to a bug reported by @nemec ([issue #244](https://github.com/commandlineparser/commandline/issues/244))

The only way to work properly, command is:

           app.exe file.hdf5  -c chanA:chanB 

Also, this is related to issue #91

**Comment by rmunn** [comment]
This will be fixed by #684 once it's merged in.
